mod/item.php

```diff
@@ -992,16 +1005,35 @@ function item_post(&$a) {
 		// Insert an item entry for UID=0 for global entries
 		// We have to remove or change some data before that,
 		// so that the post appear like a regular received post.
-		unset($datarray['self']);
-		unset($datarray['wall']);
-		unset($datarray['origin']);
-
-		if (in_array($datarray['type'], array("net-comment", "wall-comment")))
-			$datarray['type'] = 'remote-comment';
-		elseif ($datarray['type'] == 'wall')
-			$datarray['type'] = 'remote';
-
-		add_shadow_entry($datarray);
+		// Additionally there is some data that isn't a database field.
+		$arr = $datarray;
```

Comment:
What is the use of this assignation? It doesn't seem the original array `$datarray` is used further down the code, am I missing something?